jni/src/test/java/com/horizen/commitmenttree/CommitmentTreeTest.java

```diff
@@ -0,0 +1,343 @@
+        byte[] ftrPublicKeyHash = generateFieldElementBytes();
+        byte[] ftrTransactionHash = generateFieldElementBytes();
+        int fwtOutId = 200;
+        assertTrue("Forward transfer output expected to be added.",
+                commTree.addFwt(scId, ftrAmount, ftrPublicKeyHash, ftrTransactionHash, fwtOutId));
+
+        Optional<FieldElement> commitmentOpt = commTree.getCommitment();
+        assertTrue("Tree commitment expected to be present.", commitmentOpt.isPresent());
+        Optional<ScExistenceProof> existenceOpt = commTree.getScExistenceProof(scId);
+        assertTrue("Existence proof expected to be present.", existenceOpt.isPresent());
+        Optional<FieldElement> scCommitmentOpt = commTree.getScCommitment(scId);
+        assertTrue("Sidechain commitment expected to be present.", scCommitmentOpt.isPresent());
+
+        byte[] existenceProofBytes = existenceOpt.get().serialize();
+        ScExistenceProof deserializedExistanceProof = ScExistenceProof.deserialize(existenceProofBytes);
+
+        assertTrue("Deserialized existence proof should be serialized to same bytes", Arrays.equals(deserializedExistanceProof.serialize(), existenceProofBytes));
+        assertTrue("Commitment verification of original proof expected to be successful", CommitmentTree.verifyScCommitment(scCommitmentOpt.get(), existenceOpt.get(), commitmentOpt.get()));
+        assertTrue("Commitment verification of deserialized proof expected to be successful", CommitmentTree.verifyScCommitment(scCommitmentOpt.get(), deserializedExistanceProof, commitmentOpt.get()));
+        commitmentOpt.get().freeFieldElement();
+        existenceOpt.get().freeScExistenceProof();
+        scCommitmentOpt.get().freeFieldElement();
+        deserializedExistanceProof.freeScExistenceProof();
+        commTree.freeCommitmentTree();
+    }
+
+    @Test
+    public void absenceProofTest() {
+        CommitmentTree commTree = CommitmentTree.init();
+        byte[][] scId = new byte[5][];
+
+        // Initialize array of consecutive Sidechain Ids
+        for (int i = 0 ; i < scId.length; i++) {
+            scId[i] = new byte[FieldElement.getFieldElementSize()];
+            scId[i][0] = (byte) i;
+        }
+
+        Optional<FieldElement> commitmentOpt = commTree.getCommitment();
+        assertTrue("Commitment expected to be present.", commitmentOpt.isPresent());
+
+        // Get absence proof in empty CommitmentTree
+        assertFalse("Existance proof should not be present", commTree.getScExistenceProof(scId[0]).isPresent());
+        Optional<ScAbsenceProof> absenceOpt = commTree.getScAbsenceProof(scId[0]);
+        // TODO Uncomment this code when rust library will be able to operate with absence proof on empty commitment tree.
+        assertTrue("Absence proof expected to be present.", absenceOpt.isPresent());
+        assertTrue("Absence verification expected to be successful", commTree.verifyScAbsence(scId[0], absenceOpt.get() ,commitmentOpt.get()));
```

Comment:
commTree.verifyScAbsence - static member is accessed via instance reference; can be replaced with CommitmentTree.verifyScAbsence